src/lib/core/rtl/active-layout-direction.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  Injectable,
+  ElementRef,
+  ApplicationRef,
+  Injector
+} from '@angular/core';
+
+import {LayoutDirection} from './dir';
+import {EventEmitter} from '@angular/common/src/facade/async';
+
+@Injectable()
+export class ActiveLayoutDirection {
```

Comment:
Rename this class to something like `GlobalDirAccessor` and then add another class called `ActiveLayoutDirection` that just looks like
```ts
export class ActiveLayoutDirection {
  dirChange: Observable<void>;
  getActiveDirection(injector: Injector): Promise<LayoutDirection> { return null; }
}
```

The both `Dir` and `GlobalDirAccessor` will have `implements ActiveLayoutDirection` (TypeScript lets you `implement` a class just like an interface. Keeping it as a class lets us use it as an injection token).